client/app/pages/dashboards/DashboardPage.jsx

```diff
@@ -144,36 +144,48 @@ DashboardComponent.propTypes = {
   dashboard: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
 };
 
-function DashboardPage({ dashboardSlug, onError }) {
+function DashboardPage({ dashboardSlug, dashboardId, onError }) {
   const [dashboard, setDashboard] = useState(null);
   const onErrorRef = useRef();
   onErrorRef.current = onError;
 
   useEffect(() => {
-    Dashboard.get({ slug: dashboardSlug })
+    Dashboard.get({ id: dashboardId, slug: dashboardSlug })
       .then(dashboardData => {
         recordEvent("view", "dashboard", dashboardData.id);
         setDashboard(dashboardData);
       })
       .catch(error => onErrorRef.current(error));
-  }, [dashboardSlug]);
+  }, [dashboardSlug, dashboardId]);
 
   return <div className="dashboard-page">{dashboard && <DashboardComponent dashboard={dashboard} />}</div>;
 }
 
 DashboardPage.propTypes = {
-  dashboardSlug: PropTypes.string.isRequired,
+  dashboardSlug: PropTypes.string,
+  dashboardId: PropTypes.string,
   onError: PropTypes.func,
 };
 
 DashboardPage.defaultProps = {
+  dashboardSlug: null,
+  dashboardId: null,
   onError: PropTypes.func,
 };
 
+// route kept for backward compatibility
 routes.register(
   "Dashboards.ViewOrEdit",
   routeWithUserSession({
     path: "/dashboard/:dashboardSlug",
     render: pageProps => <DashboardPage {...pageProps} />,
   })
```

Comment:
Let's make this route redirect to the new one, to make sure people stop using them. If it's not easy with the frontend router, we can consider doing this on the backend.